File: CPP/PROGRAM/Ram.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class Ram : public Komponen{
    //atribut
    private:
        int kapasitasGB;
        string ddr;

    public:
        //konstruktor
        Ram(){
            this->kapasitasGB = 0;
            this->ddr = "";
        }

        Ram(int kapasitasGB, string ddr, string merk, string nama) : Komponen(merk, nama){
            this->kapasitasGB = kapasitasGB;
            this->ddr = ddr;
        }
        
        //getter dan setter
        void setKapasitasGB(int kapasitasGB){
            this->kapasitasGB = kapasitasGB;
        }

        void setDdr(string ddr){
            this->ddr = ddr;
        }

        int getKapasitasGB(){
            return this->kapasitasGB;
        }

        string getDdr(){
            return this->ddr;
        }

        ~Ram(){

        }
};

File: PYTHON/PROGRAM/Ram.py
from Komponen import Komponen

class Ram(Komponen):
    #konstruktor
    def __init__(self, merk = "", nama = "", kapasitasGB = 0, ddr = ""):
        super().__init__(merk, nama)
        self.__kapasitasGB = kapasitasGB
        self.__ddr = ddr

    #getter dan setter
    def getKapasitasGB(self):
        return self.__kapasitasGB
    
    def setKapasitasGB(self, kapasitasGB):
        self.__kapasitasGB = kapasitasGB

    def getDdr(self):
        return self.__ddr
    
    def setDdr(self, ddr):
        self.__ddr = ddr